onnxruntime/core/providers/webnn/builders/impl/squeeze_unsqueeze_op_builder.cc

```diff
@@ -78,15 +78,25 @@ Status SqueezeUnsqueezeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_buil
     }
   }
 
-  if (axes_data.size() > 0) {
-    options.set("axes", emscripten::val::array(axes_data));
-  }
-
   emscripten::val output = emscripten::val::undefined();
   if (op_type == "Squeeze") {
+    if (axes_data.size() > 0) {
+      options.set("axes", emscripten::val::array(axes_data));
+    }
     output = model_builder.GetBuilder().call<emscripten::val>("squeeze", input, options);
   } else if (op_type == "Unsqueeze") {
-    output = model_builder.GetBuilder().call<emscripten::val>("unsqueeze", input, options);
+    // Use WebNN's reshape to implement Unsqueeze.
+    std::vector<int32_t> new_shape;
+    std::transform(
+        input_shape.begin(), input_shape.end(), std::back_inserter(new_shape),
+        [](int64_t data) -> int32_t { return SafeInt<int32_t>(data); });
```

Comment:
This int64 to uint32 is repeated often. Consider adding a mini-helper function.

Reply:
This has been handled in [`HandleNegativeAxis`](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/common.h#L30).